core/state/statedb.go

```diff
@@ -736,11 +736,6 @@ func (s *StateDB) Finalise(deleteEmptyObjects bool) {
 		if obj.selfDestructed || (deleteEmptyObjects && obj.empty()) {
 			delete(s.stateObjects, obj.address)
 			s.markDelete(addr)
-
-			// If ether was sent to account post-selfdestruct it is burnt.
```

Comment:
@s1na Is it really possible that self-destructed accounts have non-zero balance?

Reply:
Sure.  The account isn't removed until the end of the transaction.  So a sub-call in the same transaction can send ether to the about-to-be-destructed account.

Reply:
Ah, right!